controllers/terminal_controller.go

```diff
@@ -236,14 +234,28 @@ func (r *TerminalReconciler) handleRequest(req ctrl.Request) (ctrl.Result, error
 		finalizers.Insert(extensionsv1alpha1.TerminalName)
 		t.Finalizers = finalizers.UnsortedList()
 
-		if err := r.Update(context.Background(), t); err != nil {
-			return ctrl.Result{}, err
-		}
+		return ctrl.Result{}, r.Update(ctx, t)
```

Comment:
Because the resource (`Terminal`) would be re-queued by this change anyways. It's better to update the resource, exit the reconciliation and then re-queue from the start. This would save some calls to the API server with Pod updates.